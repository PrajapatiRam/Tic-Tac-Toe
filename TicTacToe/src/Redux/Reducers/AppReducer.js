'use strict';
import {
  SAVE_PLAYER1_ACTION,
  SAVE_PLAYER2_ACTION,
  SAVE_PLAYER1_SCORE_ACTION,
  SAVE_PLAYER2_SCORE_ACTION,
} from '../ReduxTypes';

export default function AppReducer(state, action) {
  switch (action.type) {
    case SAVE_PLAYER1_ACTION:
      return {...state, player1name: action.payload};
    case SAVE_PLAYER2_ACTION:
      return {...state, player2name: action.payload};
    case SAVE_PLAYER1_SCORE_ACTION:
      return {...state, player1score: action.payload};
    case SAVE_PLAYER2_SCORE_ACTION:
      return {...state, player2score: action.payload};
    default:
      return state ? state : {};
  }
}
