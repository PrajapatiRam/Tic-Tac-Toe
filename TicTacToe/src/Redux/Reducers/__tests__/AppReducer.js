import mockStore from 'redux-mock-store';
import {
  savePlayer1Name,
  savePlayer2Name,
  savePlayer1Score,
  savePlayer2Score,
} from '../../Actions/AppActions';
import AppReducer from '../AppReducer';

const store = mockStore();

beforeEach(() => {
  store.clearActions();
});

it('INITIAL STATE', () => {
  const action = {type: 'DUMMY'};
  expect(AppReducer(undefined, action)).toMatchSnapshot();
});

it('savePlayer1Name: ', () => {
  expect(
    AppReducer(undefined, savePlayer1Name('Test User 1')),
  ).toMatchSnapshot();
});
it('savePlayer2Name: ', () => {
  expect(
    AppReducer(undefined, savePlayer2Name('Test User 2')),
  ).toMatchSnapshot();
});
it('savePlayer1Score: ', () => {
  expect(AppReducer(undefined, savePlayer1Score(1))).toMatchSnapshot();
});
it('savePlayer2Score: ', () => {
  expect(AppReducer(undefined, savePlayer2Score(1))).toMatchSnapshot();
});
