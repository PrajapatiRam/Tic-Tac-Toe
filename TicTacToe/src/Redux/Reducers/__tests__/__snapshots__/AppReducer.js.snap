// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL STATE 1`] = `Object {}`;

exports[`savePlayer1Name:  1`] = `
Object {
  "player1name": "Test User 1",
}
`;

exports[`savePlayer1Score:  1`] = `
Object {
  "player1score": 1,
}
`;

exports[`savePlayer2Name:  1`] = `
Object {
  "player2name": "Test User 2",
}
`;

exports[`savePlayer2Score:  1`] = `
Object {
  "player2score": 1,
}
`;
